docs/tutorial.qmd

```diff
@@ -0,0 +1,980 @@
+---
+title: "Using Flow"
+tbl-colwidths: [30,70]
+---
+
+## Core Concepts
+
+### Flow Type System
+
+Inspect Flow mirrors Inspect AI's object model with corresponding Flow types:
+
+-   **`FlowConfig`** — The top-level workflow definition that translates to an Inspect AI [Eval Set](https://inspect.aisi.org.uk/eval-sets.html). Contains a list of tasks and workflow settings.
+-   **`FlowTask`** — Configuration for a single evaluation task. Maps to Inspect AI [Task](https://inspect.aisi.org.uk/tasks.html) parameters.
+-   **`FlowModel`** — Model configuration including API settings and generation settings. Maps to Inspect AI [Model](https://inspect.aisi.org.uk/models.html).
+-   **`FlowGenerateConfig`** — Model generation parameters. Maps to Inspect AI [GenerateConfig](https://inspect.aisi.org.uk/models.html#generation-config).
+-   **`FlowSolver`** and **`FlowAgent`** — Solver and agent chain configuration. Map to Inspect AI [Solver](https://inspect.aisi.org.uk/solvers.html) and [Agent](https://inspect.aisi.org.uk/agents.html).
+-   **`FlowOptions`** — Runtime execution options like concurrency limits, error handling, and logging preferences. Maps to Inspect AI [eval_set()](https://inspect.aisi.org.uk/eval-sets.html) parameters.
+-   **`FlowDefaults`** — System for setting default values across tasks, models, solvers, and agents.
+
+### Flow Config Structure
+
+`FlowConfig` is the primary interface for defining Flow workflows. All Flow operations—including [parameter sweeps and matrix expansions](#parameter-sweeping)—ultimately produce a list of tasks that `FlowConfig` executes.
+
+**Required fields:**
+
+-   **`tasks`** — List of `FlowTask` objects defining the evaluations to run
+
+**Optional fields:**
+
+| Field | Description | Default |
+|------------------|--------------------------------|----------------------|
+| `flow_dir` | Directory for logs and resolved config. Supports S3 paths (e.g., `s3://bucket/path`) | `"logs/flow"` |
+| `python_version` | Python version for the isolated virtual environment (e.g., `"3.11"`) | Same as current environment |
+| `dependencies` | PyPI packages, Git URLs, or local paths to install | `[]` |
+| `env` | Environment variables to set in the workflow execution context | `{}` |
+| `options` | Runtime options (see `FlowOptions` reference) | `None` |
+| `defaults` | Default values applied across tasks, models, and solvers | `None` |
```

Comment:
Is there some automatic way to keep this in sync with the docstrings?

Reply:
I'll check! If so, I'll add this in a separate PR.